internal/common/distro_test.go

```diff
@@ -52,3 +52,89 @@ VARIANT_ID=workstation`,
 		}
 	}
 }
+
+func TestVersionLessThan(t *testing.T) {
+	type testCases struct {
+		Name     string
+		VersionA string
+		VersionB string
+		Expected bool
+	}
+
+	cases := []testCases{
+		{
+			Name:     "8 < 8.1",
+			VersionA: "8",
+			VersionB: "8.1",
+			Expected: true,
+		},
+		{
+			Name:     "8.1 < 8.2",
+			VersionA: "8.1",
+			VersionB: "8.2",
+			Expected: true,
+		},
+		{
+			Name:     "8 < 9",
+			VersionA: "8",
+			VersionB: "9",
+			Expected: true,
+		},
+		{
+			Name:     "8.1 < 9",
+			VersionA: "8.1",
+			VersionB: "9",
+			Expected: true,
+		},
+		{
+			Name:     "8.1 < 9.1",
+			VersionA: "8.1",
+			VersionB: "9.1",
+			Expected: true,
+		},
+		{
+			Name:     "8 < 8.10",
+			VersionA: "8",
+			VersionB: "8.10",
+			Expected: true,
+		},
+		{
+			Name:     "8.1 < 8.10",
+			VersionA: "8.1",
+			VersionB: "8.10",
+			Expected: true,
+		},
+		{
+			Name:     "8.10 < 8.11",
+			VersionA: "8.10",
+			VersionB: "8.11",
+			Expected: true,
+		},
+		{
+			Name:     "8.10 > 8.6",
+			VersionA: "8.10",
+			VersionB: "8.6",
+			Expected: false,
+		},
+		{
+			Name:     "8-stream < 9-stream",
+			VersionA: "8-stream",
+			VersionB: "9-stream",
+			Expected: true,
+		},
+		{
+			Name:     "9-stream < 9.1",
```

Comment:
This one is weird, I think that semantically, these should be incomparable. Not a biggie, though, it's just a silly cornercase.

Reply:
It is, but if you think about it more, the suffix is treated as a specifier or the version (e.g. rc or beta). Then `9` is just `9.0.0`, thus `9.1` is of higher version. But I agree that in our case and in case of CS vs RHEL versions, this does not make sense and is logically incorrect :man_shrugging:

Reply:
Would it make more sense to have a special case in the function and make it fail?

Reply:
FWIW, I'm OK with the current behavior as is. This is a situation where we technically compare versions of two different distros with different versioning scheme.

Reply:
Yeah let's merge it and if it gets weird we can change it.